src/org/jitsi/impl/neomedia/rtp/remotebitrateestimator/RemoteBitrateEstimatorSelector.java

```diff
@@ -0,0 +1,268 @@
+            }
+            ((RemoteBitrateEstimator)packetTransformer)
+                    .setMinBitrate(minBitrateBps);
+        }
+    }
+
+
+    @Override
+    public Collection<Integer> getSsrcs()
+    {
+        return ((RemoteBitrateEstimator)packetTransformer).getSsrcs();
+    }
+
+    /**
+     * Returns the estimated payload bitrate in bits per second if a valid
+     * estimate exists; otherwise, <tt>-1</tt>.
+     *
+     * @return the estimated payload bitrate in bits per seconds if a valid
+     * estimate exists; otherwise, <tt>-1</tt>
+     */
+    @Override
+    public long getLatestEstimate()
+    {
+
+        return ((RemoteBitrateEstimator)packetTransformer).getLatestEstimate();
+    }
+
+    /**
+     * Removes all data for <tt>ssrc</tt>.
+     *
+     * @param ssrc
+     */
+    @Override
+    public void removeStream(int ssrc) {
+        ((RemoteBitrateEstimator)packetTransformer).removeStream(ssrc);
+    }
+
+    @Override
+    public void setMinBitrate(int minBitrateBps) {
+        synchronized (critSect)
+        {
+            ((RemoteBitrateEstimator)packetTransformer)
+                    .setMinBitrate(minBitrateBps);
+            this.minBitrateBps = minBitrateBps;
+        }
+    }
+
+    @Override
+    public RawPacket reverseTransform(
+            RawPacket packet)
+    {
+        /**
+         * @Todo Find out why having synchronized here makes sense since it
+         * exists in PickEstimatorFromHeader.
+         */
+        if (this.absSendTimeEngine == null){
+            this.absSendTimeEngine = stream.getAbsSendTimeEngine();
+        }
+
+        synchronized (critSect)
+        {
+            if(enableUseOfAbsSendTime)
+            {
+                PickEstimatorFromHeader(packet);
+            }
+            packetTransformer.reverseTransform(packet);
+        }
+        return packet;
+    }
+
+    /**
+     * Gets the <tt>PacketTransformer</tt> for RTP packets.
+     *
+     * @return the <tt>PacketTransformer</tt> for RTP packets
+     */
+    @Override
+    public PacketTransformer getRTPTransformer() {
+        return this;
+    }
+
+    /**
+     * Gets the <tt>PacketTransformer</tt> for RTCP packets.
+     *
+     * @return the <tt>PacketTransformer</tt> for RTCP packets
+     */
+    @Override
+    public PacketTransformer getRTCPTransformer() {
+        return null;
+    }
+
+
+    /**
+     * Returns the number of milliseconds until this instance wants a worker
+     * thread to call {@link #run()}. The method is called on the same
+     * worker thread as Process will be called on.
+     *
+     * @return the number of milliseconds until this instance wants a worker
+     * thread to call {@link #run()}
+     */
+    @Override
+    public long getTimeUntilNextRun() {
+        long timeUntilNextRun  =
+                (packetTransformer instanceof RecurringRunnable) ?
+             ((RecurringRunnable) packetTransformer)
+                     .getTimeUntilNextRun() : -1;
```

Comment:
Returning -1 here will result in the executor running this immediately (there are no special semantics for -1, it just executes as long as getTimeUntilNextRun <= 0). I think the easiest solution would be to run it at the same interval as RBESingleStream (500ms). Another option would be to only register with the executor while we're using RBESingleStream, and then de-register when we switch to RBEAbsSendTime.